src/main/java/moze_intel/projecte/network/packets/SyncEmcPKT.java

```diff
@@ -30,15 +30,21 @@ public SyncEmcPKT(int[][] data)
 	public void fromBytes(ByteBuf buf)
 	{
 		int size = ByteBufUtils.readVarInt(buf, 5);
-		data = new int[size][];
+		data = new long[size][];
 
 		for (int i = 0; i < size; i++)
 		{
-			int[] array = new int[3];
+			long[] array = new long[3];
 
 			for (int j = 0; j < 3; j++)
 			{
-				array[j] = ByteBufUtils.readVarInt(buf, 5);
+				int val = ByteBufUtils.readVarInt(buf, 5);
+				if (j == 2) { //EMC it is stored in this position and the next one
+					//Pack the two ints representing the bits of the EMC long back into a long to retrieve the true value
+					array[j] = (long) val << 32 | ByteBufUtils.readVarInt(buf, 5) & 0xffffffffL;
```

Comment:
does ByteBuf and ByteBufUtils really not have methods for longs? that's surprising